src/services/audit-log-service.ts

```diff
@@ -0,0 +1,111 @@
+import Logger from "bunyan";
+import { envVars } from "config/env";
+import { getLogger } from "config/logger";
+import { dynamodb as ddb } from "config/dynamodb";
+import { createHashWithoutSharedSecret } from "utils/encryption";
+
+const defaultLogger = getLogger("DeploymentDynamoLogger");
+
+export type AuditInfoPK = {
+	entityType: string;
+	entityId: string;
+	subscriptionId: number;
+	issueKey: string;
+};
+
+export type AuditInfo = AuditInfoPK & {
+	createdAt: Date;
+	entityAction: string;
+	source: string;
+	env: string;
+};
+
+const ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+
+export const auditLog = async (auditInfo: AuditInfo, logger: Logger) => {
+	logger.debug("Saving auditInfo to db");
+	const {
+		source,
+		entityAction,
+		entityId,
+		entityType,
+		subscriptionId,
+		env,
+		issueKey,
+		createdAt
+	} = auditInfo;
+	const result = await ddb
+		.putItem({
+			TableName: envVars.DYNAMO_AUDIT_LOG_TABLE_NAME,
+			Item: {
+				Id: { S: getKey(auditInfo) }, //partition key
+				CreatedAt: { N: String(createdAt.getTime()) }, //sort key
+				ExpiredAfter: {
+					N: String(
+						Math.floor(
+							(auditInfo.createdAt.getTime() + ONE_DAY_IN_MILLISECONDS) / 1000
+						)
+					)
+				}, //ttl
+				source: { S: source },
+				entityAction: { S: entityAction },
+				entityId: { S: entityId },
+				entityType: { S: entityType },
+				env: { S: env },
+				issueKey: { S: issueKey },
+				subscriptionId: { N: String(subscriptionId) }
+			}
+		})
+		.promise();
+	if (result.$response.error) {
+		throw result.$response.error;
+	}
+};
+
+export const findLog = async (
+	params: AuditInfoPK,
+	logger: Logger = defaultLogger
+): Promise<AuditInfo | undefined> => {
```

Comment:
```suggestion
): Promise<AuditInfo[]> => {
```

Reply:
I am not sure we should return only one item or multiple. So want to make this return [] so that it is flexible.